clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -749,7 +749,10 @@ static std::string computeUniqueSYCLXOCCName(StringRef Name,
   /// Those characters need to be used wisely to prevent name collisions.
   /// It is also useful to use a name that is understandable by the user,
   /// so we add only 8 character of hash and only if needed.
+  /// The first character cannot be an underscore or a didgit.
+  /// An underscore can't be followed by an other underscore.
   constexpr unsigned MaxXOCCSize = 30;
+  bool ForceHash = false;
```

Comment:
```suggestion
  /// Some transformations might make 2 kernel identifiers the same.
  /// Allow adding a hash when such transformations are made to avoid possible name conflict.
  bool ForceHash = false;
```